### PR TITLE
health, ipn/ipnlocal: move more health warning code into health.Tracker

### PR DESCRIPTION
**Pull request by bradfitz**
In prep for making health warnings rich objects with metadata rather
than a bunch of strings, start moving it all into the same place.

We'll still ultimately need the stringified form for the CLI and
LocalAPI for compatibility but we'll next convert all these warnings
into Warnables that have severity levels and such, and legacy
stringification will just be something each Warnable thing can do.

Updates #4136

### CODE REVIEW

File: ipn/ipnlocal/profiles.go [hunk 1]
```diff
@@ -443,12 +448,20 @@ func (pm *profileManager) writeKnownProfiles() error {
 	return pm.WriteState(ipn.KnownProfilesStateKey, b)
 }
 
+func (pm *profileManager) updateHealth() {
```
Comment (bradfitz) [inline ipn/ipnlocal/profiles.go:451]: Sorry, @maisem, but this is the only way I could find to update all the bazillion paths that changed prefs in LocalBackend :(

Comment (maisem) [reply ipn/ipnlocal/profiles.go:451]: seems fine

File: health/health.go [hunk 2]
```diff
@@ -72,6 +73,9 @@ type Tracker struct {
 	watchers set.HandleSet[func(Subsystem, error)] // opt func to run if error state changes
 	timer    *time.Timer
 
+	latestVersion   *tailcfg.ClientVersion // or nil
```
Comment (maisem) [inline health/health.go:76]: maybe this should be a subsystem ¯\_(ツ)_/¯

File: health/health.go [hunk 3]
```diff
@@ -609,42 +660,76 @@ var errNetworkDown = errors.New("network down")
...
-			return t.networkErrorfLocked("haven't heard from home DERP region %v in %v", rid, d)
+			add(t.networkErrorfLocked("haven't heard from home DERP region %v in %v", rid, d))
+			return merged()
 		}
 	}
 	if t.udp4Unbound {
-		return errNoUDP4Bind
+		add(errNoUDP4Bind)
+		return merged()
 	}
```
Comment (maisem) [inline health/health.go:733]: use `add` and `merged` below too?

Comment (maisem) [reply health/health.go:733]: new line 771 is different?

Comment (maisem) [reply health/health.go:733]: oh for tests

**Review by maisem — APPROVED** [review]

**Pull request merged**